cmd/nerdctl/run_network.go

```diff
@@ -164,6 +167,51 @@ func generateNetOpts(cmd *cobra.Command, dataStore, stateDir, ns, id string) ([]
 				ports = append(ports, pm...)
 			}
 		}
+	case nettype.Container:
+		if runtime.GOOS != "linux" {
+			return nil, nil, "", nil, fmt.Errorf("currently '--network=container:<container>' can only works on linux")
+		}
+		if rootlessutil.IsRootless() {
+			return nil, nil, "", nil, fmt.Errorf("currently '--network=container:<container>' can't run rootless'")
+		}
+
+		network := strings.Split(netSlice[0], ":")
+		if len(network) != 2 {
+			return nil, nil, "", nil, fmt.Errorf("invalid network: %s, should be \"container:<id|name>\"", netSlice[0])
+		}
+		containerName := network[1]
+		client, ctx, cancel, err := newClient(cmd)
+		if err != nil {
+			return nil, nil, "", nil, err
+		}
+		defer cancel()
+		f := &containerInspector{
+			mode: "dockercompat",
+		}
+		walker := &containerwalker.ContainerWalker{
+			Client:  client,
+			OnFound: f.Handler,
+		}
+		n, err := walker.Walk(ctx, containerName)
+		if err != nil {
+			return nil, nil, "", nil, err
+		}
+		if n == 0 {
+			return nil, nil, "", nil, fmt.Errorf("no such container: %s", containerName)
+		}
+		if n > 1 {
+			return nil, nil, "", nil, fmt.Errorf("multiple IDs found with provided prefix: %s", containerName)
+		}
+		container, ok := f.entries[0].(*dockercompat.Container)
+		if !ok {
+			return nil, nil, "", nil, fmt.Errorf("expected *dockercompat.Container, got %T", f.entries[0])
+		}
+
+		opts = append(opts, oci.WithLinuxNamespace(specs.LinuxNamespace{
+			Type: specs.NetworkNamespace,
+			Path: fmt.Sprintf("/proc/%d/ns/net", container.State.Pid),
```

Comment:
IMHO, we shouldn't write directly to the network namespace, it should be joined when oci hooking

Reply:
Move to https://github.com/containerd/nerdctl/blob/master/pkg/ocihook/ocihook.go#L161

Reply:
TBH I don't really understand why we should not specify namespace here. Can you explain a little bit? 🙏

network namespace is deleted in this function as well,  when nettype is Host (https://github.com/containerd/nerdctl/blob/master/cmd/nerdctl/run_network.go#L139 ).

Reply:
OCI hook is preferable for supporting `nerdctl run --restart`

Reply:
If the reused container is restarted, the pid will change, write dead pid will not be able to restart the container

Reply:
After thinking about it, only updating the network namespace during the oci hook cannot make the container join the network.

> OCI hook is preferable for supporting `nerdctl run --restart`

It seems that docker can't do this either

Reply:
In another way, we can support `nerdctl start` join the container network.
- nerdctl run -d --name a
- nerdctl run -d --net container:a --name b
- nerdctl restart a
- nerdctl restart b (update network namespace and restart container)